### PR TITLE
Fix regression in and improve memory usage estimation

### PR DESCRIPTION
**Pull request by calebwin**

### CODE REVIEW

File: Banyan/src/investigating.jl [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# I think it's actually _good_ for this file to be check in. It's good
+# to see in each commit what was being investigated.
+
+global investigating = Dict(
```
Comment (cailinw) [inline Banyan/src/investigating.jl:4]: How is this kind of analysis different from what gets turned on with our debug mode?
or is this more granular? How does this tie in with the debug mode?

File: BanyanDataFrames/test/utils_data.jl [hunk 2]
```diff
@@ -183,6 +183,46 @@ function setup_empty_tests(bucket_name=get_cluster_s3_bucket_name())
     end
 end
 
+function setup_nyc_taxi_stress_test(nbytes = "10 GB", bucket_name=get_cluster_s3_bucket_name())
```
Comment (cailinw) [inline BanyanDataFrames/test/utils_data.jl:186]: parameterizing on nbytes is great! thanks

**Review by cailinw — APPROVED** [review]
There are several printlns, @show statements in the tests that you can probably remove.

Overall, looks great. Thanks!

File: Banyan/src/investigating.jl (hunk 1 above)
Comment (calebwin) [reply Banyan/src/investigating.jl:4]: Debug mode is for high-level visibility of Banyan client-side internals. Investigating is for investigating specific kinds of issues. There should be no overlap between things that always are for debug mode (high-level visibility) and things that are only for specific investigations.

**Pull request merged**